### ISSUE TITLE
280: Automatically copy Layers Mode properties

### ISSUE THREAD

**Issue by tngreene** [X-Plane/XPlane2Blender#483]
When we figure out how Layers Modes content will be used in 2.80, we also need to copy the contents of Layers X's layers properties.

- [x] Unit tests
   - (Probably 2 cases for cockpit and instanced scenery with a change for every property)
- [x] Code in updater
- [x] Changes to test assertions

Although LOD bucket properties will carry over, that is covered by #451.

Before we can continue with #450 we really need more unit tests. Solving this will unlock a huge amount to work with.

**Comment by tngreene** [comment]
Everything appears to be working, but, I still want to remove scene.layers and XPlaneLayer.index. For now I'm happy with how things turned out and I think it is ready for use.

**Comment by tngreene** [comment]
I did not think what would happen when you had multiple scenes: The answer when updating  `tests\\updater\\shadow\\shadow_local_off_non_scenery.test.blend'`?

```
Collection 1.001
Collection 2.001
Layer 1.001
Layer 2.001
Layer 3.001
```

I'm not exactly sure what is going on here because I'm trying to move through this quickly, but, obviously our code is looking for "Collection {number}", but not finding the match. test_creation_helpers is then creating the collections that are missing.

I think the settings will then be copied to the incorrect collection.

**Comment by tngreene** [comment]
Another issue:

We must not set converted layers as exportable if root objects mode was chosen or if objects inside that layer are marked as root objects. Otherwise "nested roots" error is raised.

3 notes:
- `unmark {name} as a Root or change its parentage`, what if the parentage is a root
- "objects inside that layer are marked as root objects". The theory here of making this the default is we are going forward with Root Objects only and it takes more work in 2.79 to make this happen. We can put a note in the updater log.
- [ ] We need to distribute these notes across another bug I think...

**Comment by tngreene** [comment]
Also! Update paths that involve asking "scene.xplane.layers[0].something" need this copying done before hand.

This entails a much better update operation. We need to do the layers copying first and re-write previous update code to talk about scene.collection.children["Layer 1"].xplane.layer instead!

**Comment by tngreene** [comment]
So, after much aggrevation and a surprisingly long amount of time the feature is finished. I am concerned, however, about the following.

Suppose the following assumption:
```
A blend file has files A, B, and C.

A has something on Layer 2, Collection 2 gets made By Blender
B has something on Layer 2, Collection 2.001 gets made By Blender
C has something on Layer 2, Collection 2.002 gets made by Blender
```
Now we assume that 3rd scene = C = Blender's second chance to make a collection with a conflicting name (Collection 2.001). Everything lines up nice and neat.

Now suppose that isn't true, and the order of iteration will be A, C, B. Does this make a difference? It might for the unit test. I'm running out of mental power to think about this. For now, however, we're closed.

**Comment by tngreene** [comment]
Closed with 3cf681a9b4ac0
